search/src/Components/upload/upload_stepper.js

```diff
@@ -251,9 +259,9 @@ class UploadStepper extends React.Component {
                     <Button
                         variant="contained"
                         color="primary"
-                        onClick={this.handleNext}
+                        onClick={this.handleAgContextExempt() ? this.handleUploadAgcontexts : this.handleNext}
```

Comment:
This is starting to get messy... too much specialisation in generic code.

The correct way to do it is for each step to have an optional handleNext hook that can decide whether to continue to the default behaviour.

I'm fine to leave the correct way until later. (Open an issue?) :)